internal/util/connection.go

```diff
@@ -71,7 +71,13 @@ func (cc *ClusterConnection) GetIoctx(pool string) (*rados.IOContext, error) {
 
 	ioctx, err := cc.conn.OpenIOContext(pool)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to open IOContext for pool %s", pool)
+		// ErrNotFound indicates the Pool was not found
+		if err == rados.ErrNotFound {
```

Comment:
@nixpanic are there any references available to the `rados` error codes ?

Reply:
If yes, it would be good to mention here.

Reply:
Not much, they are just `errno` values converted to Golang errors. The go-ceph functions do not specify what errors can be returned, as that depends directly on the Ceph libraries (and these do not document it either).

Reply:
@nixpanic sure, I was confused by this ` 		// ErrNotFound indicates the Pool was not found ` . Is it true that, `ErrNotFound` is the error code  just for `Pool was not found`, If that's the case , I am wondering why it was not used as `ErrPoolNotFound` in first place?

Reply:
`ErrNotFound` is a translation of the errno `ENOENT` and can indicate any missing resource. When opening a pool, a  return of `ENOENT` is most likely referring to the non-existence of the pool. There is no `ErrPoolNotFound` in go-ceph/rados/errors.go.

Reply:
@nixpanic oh.. my understanding was that it could be any failure like `osds` are down or  `PG related` or `node down`..etc. Isn't it the case?

Reply:
in that case there would be some connection error

Reply:
@nixpanic so you are confirming we wont get into the openContext at all or we would have failed early, Isn't it?